### PR TITLE
refactor(examples): extend examples printers

### PR DESCRIPTION
**Pull request by msmk0**
Extend the printers algorithms to be able to print particles, hits, and track parameters. The reading example for CSV files is replaced by two utilties, `ActsExampleShowParticles` and `ActsExampleShowFatrasGeneric`. They either print just particles or the full simulated Fatras output using the updated printer algorithms. Contains the following related changes:

- Require file stems to be explicitly set in the Csv particle reader and writer. This avoids cases, where multiple particle files exist for each event and there is no well-defined default one.
- Use a more compact `Particle` output.
- Print the Fatras process type by name instead of just a number.
- Remove the (recently introduced) brackets from the `GeometryIdentifier` and `MultiIndex` output. They were introduced to better separate the id from other information in a combined output. After using it for while, I realize that it is just distracting and possible separation is better done at a different level.

  Example: the particle output prints an identifier encoded by an `MultiIndex` as `id=(x|y|z)|pdg=...`. Without the brackets this would end up as `id=x|y|z|pdg=...`. The bracketing is now handled by the `Particle` output, not by the `MultiIndex`.

### CODE REVIEW

File: Examples/Algorithms/Printers/ActsExamples/Printers/HitsPrinter.cpp [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+
+#include <vector>
+
+ActsExamples::HitsPrinter::HitsPrinter(
+    const ActsExamples::HitsPrinter::Config& cfg, Acts::Logging::Level level)
+    : BareAlgorithm("HitsPrinter", level), m_cfg(cfg) {
+  if (m_cfg.inputClusters.empty()) {
+    throw std::invalid_argument("Input clusters collection is not configured");
+  }
+  if (m_cfg.inputClusters.empty()) {
```
Comment (XiaocongAi) [inline Examples/Algorithms/Printers/ActsExamples/Printers/HitsPrinter.cpp:27]: The `m_cfg.inputClusters` should be `m_cfg.inputHitParticlesMap`?

File: Examples/Run/Show/ShowParticles.cpp [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+      std::make_shared<CsvParticleReader>(readParticlesCfg, logLevel));
+
+  // pre-select particles
+  auto selectParticlesCfg = ParticleSelector::readConfig(vars);
+  selectParticlesCfg.inputParticles = readParticlesCfg.outputParticles;
+  selectParticlesCfg.outputParticles = "particles_selected";
+  sequencer.addAlgorithm(
+      std::make_shared<ParticleSelector>(selectParticlesCfg, logLevel));
+
+  // print event data
```
Comment (XiaocongAi) [inline Examples/Run/Show/ShowParticles.cpp:63]: Does `print selected particles` sound better?

**Review by XiaocongAi — COMMENTED** [review]
Hi @msmk0 , the PR looks quite good to me! I put a few minor comments and will approve once they are addressed.

File: Examples/Algorithms/Printers/ActsExamples/Printers/HitsPrinter.cpp [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+  if (0 < m_cfg.selectIndexLength) {
+    size_t ihit = m_cfg.selectIndexStart;
+    size_t nend = std::min(clusters.size(), ihit + m_cfg.selectIndexLength);
+
+    if (nend <= ihit) {
+      ACTS_WARNING("event "
+                   << ctx.eventNumber << " collection '" << m_cfg.inputClusters
+                   << " hit index selection is outside the available range");
+    } else {
+      ACTS_INFO("event " << ctx.eventNumber << " hits by index selection");
```
[the remark this answers is not in the document]
Comment (msmk0) [reply Examples/Algorithms/Printers/ActsExamples/Printers/HitsPrinter.cpp]: Yes, I did. Fixed.

File: Examples/Algorithms/Printers/ActsExamples/Printers/HitsPrinter.cpp [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+
+      Acts::GeometryIdentifier prevGeoId;
+      for (; ihit < nend; ++ihit) {
+        auto hitId = hitIds.at(ihit);
+        auto ic = clusters.nth(ihit);
+        if (ic == clusters.end()) {
```
[the remark this answers is not in the document]
Comment (msmk0) [reply Examples/Algorithms/Printers/ActsExamples/Printers/HitsPrinter.cpp]: Yes, you are right. I removed this check and added a comment.

File: Examples/Algorithms/Printers/ActsExamples/Printers/HitsPrinter.cpp (hunk 1 above)
Comment (msmk0) [reply Examples/Algorithms/Printers/ActsExamples/Printers/HitsPrinter.cpp]: Yes, it should. Thanks for catching that. Fixed.

File: Examples/Run/Show/ShowParticles.cpp (hunk 2 above)
Comment (msmk0) [reply Examples/Run/Show/ShowParticles.cpp]: Changed.

**Comment by msmk0** [comment]
@XiaocongAi I hope all your comments were properly addressed.

**Review by XiaocongAi — APPROVED** [review]
@msmk0 , thanks! Let's get it in.

**Pull request merged**